client/finality-grandpa/src/authority_set_changes.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use parity_scale_codec::{Decode, Encode};
+use std::{cmp::Ord, sync::Arc};
+
+// Tracks authority set changes. We store the block numbers for the last block of each authority
+// set.
+#[derive(Debug, Encode, Decode)]
+pub(crate) struct AuthoritySetChanges<N> {
```

Comment:
Maybe this can go inside `AuthoritySet`? We will always be operating on both at the same time, i.e. when we finalize something from `AuthoritySet` is also when we update this.